### PR TITLE
docs: add architecture draft

### PR DESCRIPTION
**Pull request by jecluis**
Signed-off-by: Joao Eduardo Luis \<joao@suse.com>

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+```
+
+### From nothing to a single node
+
+1. User flashes image on a given box, and boots;
+2. Upon boot, backend starts;
+3. Backend serves frontend;
+4. On input, backend bootstraps minimal cluster;
+5. Backend provides information to frontend on state of deployment upon request;
+6. Once bootstrapped, backend obtains hosts informations from mgr orch
```
[the remark this answers is not in the document]
Comment (ml8mr) [reply docs/architecture.md:18]:
```suggestion
6. Once bootstrapped, backend obtains host's information from mgr orch
```

**Review by ml8mr — APPROVED** [review]

**Comment by asettle** [comment]
@jecluis how much more work on this do you see being required before we can merge this? This is a living document regardless.

**Comment by jecluis** [comment]
> @jecluis how much more work on this do you see being required before we can merge this? This is a living document regardless.

pretty sure most of this will have to be rewritten by now.

**Comment by asettle** [comment]
> > @jecluis how much more work on this do you see being required before we can merge this? This is a living document regardless.
> 
> pretty sure most of this will have to be rewritten by now.

What joy.

**Comment by jecluis** [comment]
This is stale and would need a considerable (if not full) rewrite. Let's try doing that properly with another attempt later.

**Pull request closed without merging**